### PR TITLE
Upgrade to v0.7.0

### PR DESCRIPTION
**Pull request by lapineige**
A new version 0.7.0 is available https://github.com/superseriousbusiness/gotosocial/releases/tag/v0.7.0

## PR Status

- [ ] Code finished and ready to be reviewed/tested
- [ ] The fix/enhancement were manually tested (if applicable)

## Automatic tests

Automatic tests can be triggered on https://ci-apps-dev.yunohost.org/ *after creating the PR*, by commenting "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!". (N.B. : for this to work you need to be a member of the Yunohost-Apps organization)

### CODE REVIEW

**Comment by lapineige** [comment]
!testme

**Comment by lapineige** [comment]
> 278436 INFO WARNING - 2023/02/06 11:19:17 error executing command: error creating dbservice: db migration error: ERROR: relation "shortcodedomain" already exists (SQLSTATE 42P07)

I wonder if the migration can go well in a fresh new install 🤔

@tsmethurst can I ask for your input here ?
According to the log of our CI, install fails because of a migration error : https://ci-apps-dev.yunohost.org/ci/job/5518
I have no clue :(

**Comment by OniriCorpe** [comment]
Hi @lapineige!

I chose not to package the RC versions because YunoHost packages need a minimum of stability, which avoids having n panicked users for each release
Even if it's an alpha software and users are warned about it, I think it's always better to wait a few weeks for the final versions rather than releasing the RC to YunoHost installations 😅

Btw, that's why the automatic update script doesn't do PR in case of RC:
https://github.com/YunoHost-Apps/gotosocial_ynh/blob/b04e75bf71d67c4717adb7acc00ff5e7b166e06a/.github/workflows/updater.sh#L44-L47

**Comment by lapineige** [comment]
>  than releasing the RC to YunoHost installations 

I'm not willing to release it ! :)
But that could live in testing (or even in this branch) until the final version is out. Then the work to adapt to the final release should be really limited :)
I'm just anticipating the work for the final release, but there is no hurry :)

> Btw, that's why the automatic update script doesn't do PR in case of RC:

What does this script does ? Is it the auto-updater ?

**Comment by OniriCorpe** [comment]
okay I see, that's a good idea ^w^

> What does this script does ? Is it the auto-updater ?

yes it's a script that is triggered every day and if there is a new version released, it will open a PR and modify the `.src` files as necessary!

it launches the CI test and you have to check the changelog upstream and if everything is OK you just have to validate the PR, this is pretty convenient

**Comment by lapineige** [comment]
!testme

**Comment by lapineige** [comment]
Lots of bugs fixed in this RC2, not yet all of them as reported in the announcement.

**Comment by alexAubin** [comment]
(I'm deleting the previous comments from yunohost-bot because we're moving the ci-apps-dev and could be that the job ID corresponds to in fact another app ... Please retrigger the job in a couple minutes / hours, sorry for the inconvenience :sweat_smile: )

**Comment by alexAubin** [comment]
!testme

**Comment by lapineige** [comment]
Lots of CI trouble recently, I hope it's not too much to handle for the team :sweat_smile: :hugs:
We have time for this one :)

**Comment by lapineige** [comment]
```
107878 INFO DEBUG - + /var/www/gotosocial/gotosocial --config-path /var/www/gotosocial/config.yaml admin account create --username package_checker --email user@example.com --password **********
107879 INFO DEBUG - timestamp="13/02/2023 08:17:17.522" func=bundb.pgConn level=INFO msg="connected to POSTGRES database"
107879 INFO WARNING - WARN: bun: 2023/02/13 08:17:21 query "remote_url" has [count(*)] args, but no placeholders
107879 INFO DEBUG - timestamp="13/02/2023 08:17:21.886" func=migrations.init.8.func2.1 level=INFO migration=20220612091800_duplicated_media_cleanup msg="found 0 attachments with duplicate remote URLs"
107879 INFO WARNING - 2023/02/13 08:17:22 error executing command: error creating dbservice: db migration error: ERROR: relation "shortcodedomain" already exists (SQLSTATE 42P07)
```
Wtf ? :thinking:

**Comment by lapineige** [comment]
And now we go for RC3 :D

**Comment by lapineige** [comment]
!testme

**Comment by lapineige** [comment]
!testme

**Comment by lapineige** [comment]
So with all that RC preparation work, I know it will fail :joy:

**Comment by lapineige** [comment]
So...
> 69495 INFO INFO - [################....] > Creating gotosocial admin user... [00h00m,01s]
69495 INFO DEBUG - timestamp="17/02/2023 10:08:11.150" func=bundb.pgConn level=INFO msg="connected to POSTGRES database"
69495 INFO WARNING - WARN: bun: 2023/02/17 10:08:12 query "remote_url" has [count(*)] args, but no placeholders
69496 INFO DEBUG - timestamp="17/02/2023 10:08:12.184" func=migrations.init.8.func2.1 level=INFO migration=20220612091800_duplicated_media_cleanup msg="found 0 attachments with duplicate remote URLs"
69496 INFO WARNING - 2023/02/17 10:08:12 error executing command: error creating dbservice: db migration error: ERROR: relation "shortcodedomain" already exists (SQLSTATE 42P07)

Does it dislike the fact that we already created a PostgrSQL database ?
Any idea ? 😅

**Comment by OniriCorpe** [comment]
maybe it's because of that?

> config.yaml
> 
> The configuration file has changed since the previous release. We recommend copying the new file from example/config.yaml and pasting values into it from your previous config.yaml. You can see a diff of the config file here: https://github.com/superseriousbusiness/gotosocial/compare/v0.6.0...v0.7.0#diff-c071e03510b2c57e193a44503fd9528a785f0f411497cc75841a9f8d0b1ac622

**Comment by lapineige** [comment]
Good catch !!

**Comment by lapineige** [comment]
!testme

File: conf/config.yaml [hunk 1]
```diff
@@ -594,3 +646,41 @@ advanced-cookies-samesite: "__ADVANCED_COOKIES_SAMESITE__"
 # Examples: [1000, 500, 0]
 # Default: 1000
 advanced-rate-limit-requests: __ADVANCED_RATE_LIMIT_REQUESTS__
+
+# Default: 300
+advanced-rate-limit-requests: 300
```
Comment (OniriCorpe) [inline conf/config.yaml:651]: `advanced-rate-limit-requests` is now duplicated :/

**Review by OniriCorpe — CHANGES_REQUESTED** [review]

**Comment by lapineige** [comment]
!testme

**Comment by lapineige** [comment]
Same error :(

**Comment by OniriCorpe** [comment]
idk :(

I asked for help in the matrix chat of the GTS project:

> Hi!
> It's been a long time ^w^
> 
> We have an error for the last version on the yunohost packaging
> 
> 67913 WARNING 2023/02/17 10:08:12 error executing command: error creating dbservice: db migration error: ERROR: relation "shortcodedomain" already exists (SQLSTATE 42P07)
> 
> https://github.com/YunoHost-Apps/gotosocial_ynh/pull/63

https://matrix.to/#/!mlPctfMHjyopbOnliM:superseriousbusiness.org/$qcCV08Im52qiBxbrWy-_GgAvjdsoa-AXWnTvrs0BkYk?via=superseriousbusiness.org&via=matrix.org&via=tchncs.de

**Review by OniriCorpe — APPROVED** [review]

**Comment by OniriCorpe** [comment]
the bug has been fixed and will be available in v0.7.1 🎉 
https://github.com/superseriousbusiness/gotosocial/pull/1528

**Comment by tsmethurst** [comment]
> the bug has been fixed and will be available in v0.7.1

sorry i'm so late to the party! I need repeated poking to actually get stuff done, these days :')

**Comment by OniriCorpe** [comment]
> in future you're better messaging me via matrix

will do ❤️
thank you so much

**Comment by tsmethurst** [comment]
No, thank you! :)

**Comment by lapineige** [comment]
> > the bug has been fixed and will be available in v0.7.1
> 
> sorry i'm so late to the party! I need repeated poking to actually get stuff done, these days -- in future you're better messaging me via matrix if there's an issue with the yunohost packaging, rather than tagging me in github; i wanna support the efforts of you folks, so please poke me again if anything else comes up!

Thanks for the advice, it will be useful in the future.
But for this upgrade there was no hurry so it wasn't necessary :)

**Comment by lapineige** [comment]
> the bug has been fixed and will be available in v0.7.1 tada [superseriousbusiness/gotosocial#1528](https://github.com/superseriousbusiness/gotosocial/pull/1528)

So we wait for 0.7.1 ? :)

**Comment by OniriCorpe** [comment]
this version will never work on ynh, so I close this PR

I made the changes that came with this version on [another PR](https://github.com/YunoHost-Apps/gotosocial_ynh/pull/65) (and the changes implied in the panel configuration) so it will be ready for the next version

**Comment by lapineige** [comment]
Oh, that's why it's conflicting with testing :)

> (and the changes implied in the panel configuration)

What do you mean ?

**Comment by OniriCorpe** [comment]
it is a configuration panel accessible from the yunohost administration web interface

here's a screeshot:

a screenshot of the GoToSocial YNH config panel

**Comment by lapineige** [comment]
Oh that why so many values are configurable in the config.yaml file !

**Comment by OniriCorpe** [comment]
yep!

**Pull request closed without merging**